sources/packages/backend/apps/api/src/route-controllers/note/_tests_/e2e/note.institutions.controller.getStudentNotes.e2e-spec.ts

```diff
@@ -0,0 +1,238 @@
+
+    // Act/Assert
+    return request(app.getHttpServer())
+      .get(`/institutions/note/student/${student.id}`)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({});
+  });
+
+  it("Should get specific student notes types when student has notes and a note type filter was provided.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+
+    const student = savedApplication.student;
+    const user = await db.user.save(createFakeUser());
+    const expectedNote = createFakeNote(NoteType.Application);
+    await saveFakeStudentNotes(
+      db.dataSource,
+      [
+        createFakeNote(NoteType.General),
+        expectedNote,
+        createFakeNote(NoteType.Designation),
+      ],
+      student.id,
+      user,
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    return request(app.getHttpServer())
+      .get(
+        `/institutions/note/student/${student.id}?noteType=${NoteType.Application}`,
+      )
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([noteToApiReturn(expectedNote)]);
+  });
+
+  it("Should not get notes for student restriction with notification type 'no effect' when any are available.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    const student = savedApplication.student;
+    // Get any restriction with notification type "No effect".
+    const notificationTypeNoEffectRestriction = await db.restriction.findOne({
+      where: {
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    });
+    await saveFakeStudentRestriction(db.dataSource, {
+      student,
+      restriction: notificationTypeNoEffectRestriction,
+      application: savedApplication,
+    });
+    const user = await db.user.save(createFakeUser());
+    const [generalNote, designationNote] = await saveFakeStudentNotes(
+      db.dataSource,
+      [createFakeNote(NoteType.Designation), createFakeNote(NoteType.General)],
+      student.id,
+      user,
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    return request(app.getHttpServer())
+      .get(`/institutions/note/student/${student.id}`)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([noteToApiReturn(designationNote), noteToApiReturn(generalNote)]);
+  });
+
+  function noteToApiReturn(note: Note) {
+    return {
+      noteType: note.noteType,
+      description: note.description,
+      firstName: note.creator.firstName,
+      lastName: note.creator.lastName,
+      createdAt: note.createdAt.toISOString(),
+    };
+  }
+
```

Comment:
Scenarios missing for HTTP 403 retuned for a BC Public institution and accessing student data who has not submitted an application to the institution.

Reply:
In that case the decorator `@HasStudentDataAccess("studentId")` will throw a ForbiddenException.
I changed to:
    √ Should throw Forbidden when student does not exist.                                                                   
    √ Should throw Forbidden when student exists but does not have an application for the institution.